### ISSUE TITLE
NaiveLCAFinder#getLCASet finds not only LCAs but also additional common ancestors

### ISSUE THREAD

**Issue by miriSch** [jgrapht/jgrapht#953]
```
 * JGraphT version: 1.5.0
 * Java version (java -version)/platform:  java 11 (openjdk) / Linux Mint 19 64 bit
```

**Issue**
NaiveLCAFinder#getLCASet finds not only LCAs but also additional common ancestors

**Steps to reproduce (small coding example)**
```
public static void main(String[] args) {
		Graph<String, DefaultEdge> graph = GraphTypeBuilder.<String, DefaultEdge>directed().allowingMultipleEdges(false)
				.allowingSelfLoops(false).weighted(false).edgeSupplier(SupplierUtil.createDefaultEdgeSupplier())
				.buildGraph();
    	        /*  
		 *       a-->b-->c
		 *       |       ^
		 *       V       |
		 *       d-->e-->f
		 * 
    	         */
		String a = "a";
		String b = "b";
		String c = "c";
		String d = "d";
		String e = "e";
		String f = "f";
		graph.addVertex(a);
		graph.addVertex(b);
		graph.addVertex(c);
		graph.addVertex(d);
		graph.addVertex(e);
		graph.addVertex(f);
		graph.addEdge(a, b);
		graph.addEdge(b, c);
		graph.addEdge(a, d);
		graph.addEdge(f, c);
		graph.addEdge(d, e);
		graph.addEdge(e, f);

		NaiveLCAFinder<String, DefaultEdge> naiveLcaFinder = new NaiveLCAFinder<String, DefaultEdge>(graph);
		Set<String> lcaSet = naiveLcaFinder.getLCASet(c, e);

		System.out.println(lcaSet);
	}
```

**Expected behaviour**
The common ancestors of c and e are {a,d,e}. the only leaf of this subgraph is e, so I would expect the program to print [e] instead of [a, e]

**Other information**
The determined "intersection" set should be {a,d,e} but is only {a,e}. Because a and e are not directly connected, the leaf detection does not find that a is no LCA

**Comment by jsichi** [comment]
Confirmed that the current implementation produces the observed result.  And I agree that the observed result is incorrect according to the specification at the beginning of the Javadoc.

**Comment by d-michail** [comment]
Fixed in #963
